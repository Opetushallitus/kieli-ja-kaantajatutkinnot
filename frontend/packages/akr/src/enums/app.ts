export enum AppConstants {
  CallerID = '1.2.246.562.10.00000000001.akr',
}

export enum AppRoutes {
  PublicHomePage = '/akr/etusivu',
  ClerkHomePage = '/akr/virkailija',
  MeetingDatesPage = '/akr/virkailija/kokouspaivat',
  ExaminationDatesPage = '/akr/virkailija/tutkintopaivat',
  ClerkSendEmailPage = '/akr/virkailija/laheta-sahkoposti',
  ClerkLocalLogoutPage = '/akr/cas/localLogout',
  ClerkTranslatorOverviewPage = '/akr/virkailija/kaantaja/:translatorId',
  ClerkNewTranslatorPage = '/akr/virkailija/lisaa-kaantaja',
  AccessibilityStatementPage = '/akr/saavutettavuusseloste',
  PrivacyPolicyPage = '/akr/tietosuojaseloste',
  NotFoundPage = '*',
}

export enum HeaderTabNav {
  Register = 'register',
  ExaminationDates = 'examinationDates',
  MeetingDates = 'meetingDates',
}

export enum PermissionToPublish {
  Yes = 'yes',
  No = 'no',
}

export enum PublicUIViews {
  PublicTranslatorListing = 'PublicTranslatorListing',
  ContactRequest = 'ContactRequest',
}

export enum SearchFilter {
  FromLang = 'fromLang',
  ToLang = 'toLang',
  Name = 'name',
  Town = 'town',
}

export enum UIMode {
  EditAuthorisationDetails = 'editAuthorisationDetails',
  EditTranslatorDetails = 'editTranslatorDetails',
  View = 'view',
}
