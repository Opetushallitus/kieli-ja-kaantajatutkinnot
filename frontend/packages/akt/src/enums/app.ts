export enum AppConstants {
  CallerID = '1.2.246.562.10.00000000001.akt',
}

export enum AppRoutes {
  PublicHomePage = '/akt/etusivu',
  ClerkHomePage = '/akt/virkailija',
  MeetingDatesPage = '/akt/virkailija/kokouspaivat',
  ClerkSendEmailPage = '/akt/virkailija/laheta-sahkoposti',
  ClerkLocalLogoutPage = '/akt/cas/localLogout',
  ClerkTranslatorOverviewPage = '/akt/virkailija/kaantaja/:translatorId',
  ClerkNewTranslatorPage = '/akt/virkailija/lisaa-kaantaja',
  AccessibilityStatementPage = '/akt/saavutettavuusseloste',
  NotFoundPage = '*',
}

export enum Duration {
  Short = 3000,
  Medium = 6000,
  MediumExtra = 9000,
  Long = 12000,
}

export enum SearchFilter {
  FromLang = 'fromLang',
  ToLang = 'toLang',
  Name = 'name',
  Town = 'town',
}

export enum KeyboardKey {
  Enter = 'Enter',
}

export enum PublicUIViews {
  PublicTranslatorListing = 'PublicTranslatorListing',
  ContactRequest = 'ContactRequest',
}

export enum TextFieldTypes {
  Text = 'text',
  Email = 'email',
  PhoneNumber = 'tel',
  Textarea = 'textarea',
  PersonalIdentityCode = 'personalIdentityCode',
}

export enum CustomTextFieldErrors {
  Required = 'errors.customTextField.required',
  MaxLength = 'errors.customTextField.maxLength',
  EmailFormat = 'errors.customTextField.emailFormat',
  TelFormat = 'errors.customTextField.telFormat',
  PersonalIdentityCodeFormat = 'errors.customTextField.personalIdentityCodeFormat',
}

export enum NotifierTypes {
  Dialog = 'dialog',
  Toast = 'toast',
}

export enum Severity {
  Info = 'info',
  Success = 'success',
  Error = 'error',
  Warning = 'warning',
}

export enum Variant {
  Text = 'text',
  Outlined = 'outlined',
  Contained = 'contained',
}

export enum TextFieldVariant {
  Filled = 'filled',
  Outlined = 'outlined',
  Standard = 'standard',
}

export enum SkeletonVariant {
  Text = 'text',
  Rectangular = 'rectangular',
  Circular = 'circular',
}

export enum I18nNamespace {
  Common = 'common',
  Translation = 'translation',
  KoodistoLanguages = 'koodistoLanguages',
  Accessibility = 'accessibility',
}

export enum Color {
  Primary = 'primary',
  Secondary = 'secondary',
  Inherit = 'inherit',
  Error = 'error',
  Warning = 'warning',
}

export enum HTTPStatusCode {
  Ok = 200,
  Created = 201,
  Accepted = 202,
  NoContent = 204,
  MovedPermanently = 301,
  Found = 302,
  NotModified = 304,
  TemporaryRedirect = 307,
  PermanentRedirect = 308,
  BadRequest = 400,
  Unauthorized = 401,
  Forbidden = 403,
  NotFound = 404,
  MethodNotAllowed = 405,
  InternalServerError = 500,
  NotImplemented = 501,
  BadGateway = 502,
  ServiceUnavailable = 503,
  GatewayTimeout = 504,
}

export enum Screenwidth {
  Phone = 480,
  Tablet = 800,
  Desktop = 1024,
}

export enum PermissionToPublish {
  Yes = 'Kyll√§',
  No = 'Ei',
}

export enum AppLanguage {
  Finnish = 'fi-FI',
  Swedish = 'sv-SE',
  English = 'en-GB',
}

export enum Direction {
  Horizontal = 'horizontal',
  Vertical = 'vertical',
}

export enum UIMode {
  View = 'view',
  EditTranslatorDetails = 'editTranslatorDetails',
  EditAuthorisationDetails = 'editAuthorisationDetails',
}

export enum HeaderTabNav {
  Register = 'register',
  MeetingDates = 'meetingDates',
}
