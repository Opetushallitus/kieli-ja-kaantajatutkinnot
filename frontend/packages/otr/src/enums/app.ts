export enum AppConstants {
  CallerID = '1.2.246.562.10.00000000001.otr',
}

export enum AppRoutes {
  PublicHomePage = '/otr/etusivu',
  AccessibilityStatementPage = '/otr/saavutettavuusseloste',
  NotFoundPage = '*',
}

export enum SearchFilter {
  FromLang = 'fromLang',
  ToLang = 'toLang',
  Name = 'name',
  Area = 'area',
}

export enum PublicUIViews {
  PublicTranslatorListing = 'PublicTranslatorListing',
  ContactRequest = 'ContactRequest',
}

export enum CustomTextFieldErrors {
  Required = 'errors.customTextField.required',
  MaxLength = 'errors.customTextField.maxLength',
  EmailFormat = 'errors.customTextField.emailFormat',
  TelFormat = 'errors.customTextField.telFormat',
  PersonalIdentityCodeFormat = 'errors.customTextField.personalIdentityCodeFormat',
}

// TODO: check this
export enum UIMode {
  View = 'view',
  EditTranslatorDetails = 'editTranslatorDetails',
  EditAuthorisationDetails = 'editAuthorisationDetails',
}
