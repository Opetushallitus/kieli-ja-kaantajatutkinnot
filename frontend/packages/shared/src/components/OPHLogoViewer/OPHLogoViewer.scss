.oph-logo {
  display: block;
  margin: 0.5rem 1px;
  text-decoration: none;

  &__img {
    margin-right: 2.5rem;
  }

  &__label {
    &#{&} {
      line-height: 1rem;
    }

    &--bold {
      &#{&} {
        margin-left: 2.5rem;
        font-weight: bold;
      }
    }
  }
}
