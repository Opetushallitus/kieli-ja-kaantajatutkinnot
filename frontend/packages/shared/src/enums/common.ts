export enum APIResponseStatus {
  NotStarted = 'NOT_STARTED',
  InProgress = 'IN_PROGRESS',
  Success = 'SUCCESS',
  Error = 'ERROR',
  Cancelled = 'CANCELLED',
}

export enum AppLanguage {
  Finnish = 'fi-FI',
  Swedish = 'sv-SE',
  English = 'en-GB',
}

export enum Color {
  Primary = 'primary',
  Secondary = 'secondary',
  Inherit = 'inherit',
  Error = 'error',
  Warning = 'warning',
}

export enum CustomTextFieldErrors {
  Required = 'errors.customTextField.required',
  MaxLength = 'errors.customTextField.maxLength',
  EmailFormat = 'errors.customTextField.emailFormat',
  TelFormat = 'errors.customTextField.telFormat',
  PersonalIdentityCodeFormat = 'errors.customTextField.personalIdentityCodeFormat',
}

export enum Direction {
  Horizontal = 'horizontal',
  Vertical = 'vertical',
}

export enum Duration {
  Short = 3000,
  Medium = 6000,
  MediumExtra = 9000,
  Long = 12000,
}

export enum HTTPStatusCode {
  Ok = 200,
  Created = 201,
  Accepted = 202,
  NoContent = 204,
  MovedPermanently = 301,
  Found = 302,
  NotModified = 304,
  TemporaryRedirect = 307,
  PermanentRedirect = 308,
  BadRequest = 400,
  Unauthorized = 401,
  Forbidden = 403,
  NotFound = 404,
  MethodNotAllowed = 405,
  InternalServerError = 500,
  NotImplemented = 501,
  BadGateway = 502,
  ServiceUnavailable = 503,
  GatewayTimeout = 504,
}

export enum I18nNamespace {
  Accessibility = 'accessibility',
  Common = 'common',
  KoodistoLanguages = 'koodistoLanguages',
  KoodistoRegions = 'koodistoRegions',
  Privacy = 'privacy',
  Translation = 'translation',
}

export enum KeyboardKey {
  Enter = 'Enter',
}

export enum NotifierTypes {
  Dialog = 'dialog',
  Toast = 'toast',
}

export enum Screenwidth {
  Phone = 480,
  Tablet = 800,
  Desktop = 1024,
}

export enum Severity {
  Info = 'info',
  Success = 'success',
  Error = 'error',
  Warning = 'warning',
}

export enum SkeletonVariant {
  Text = 'text',
  Rectangular = 'rectangular',
  Circular = 'circular',
}

export enum TextFieldTypes {
  Text = 'text',
  Email = 'email',
  PhoneNumber = 'tel',
  Textarea = 'textarea',
  PersonalIdentityCode = 'personalIdentityCode',
}

export enum TextFieldVariant {
  Filled = 'filled',
  Outlined = 'outlined',
  Standard = 'standard',
}

export enum Variant {
  Text = 'text',
  Outlined = 'outlined',
  Contained = 'contained',
}
