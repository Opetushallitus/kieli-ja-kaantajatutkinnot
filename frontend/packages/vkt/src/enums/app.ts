export enum AppConstants {
  CallerID = '1.2.246.562.10.00000000001.vkt',
}

export enum AppRoutes {
  PublicHomePage = '/vkt/etusivu',
  ClerkHomePage = '/vkt/virkailija',
  ClerkExamEventPage = '/vkt/virkailija/tutkintotilaisuus/:examEventId',
  ClerkLocalLogoutPage = '/vkt/cas/localLogout',
  AccessibilityStatementPage = '/vkt/saavutettavuusseloste',
  PrivacyPolicyPage = '/vkt/tietosuojaseloste',
  NotFoundPage = '*',
}

export enum PublicUIViews {
  Enrollment = 'Enrollment',
  EnrollmentComplete = 'EnrollmentComplete',
  ExamEventListing = 'ExamEventListing',
}

export enum ExamLanguage {
  ALL = 'ALL',
  FI = 'FI',
  SV = 'SV',
}

export enum ExamLevel {
  EXCELLENT = 'EXCELLENT',
}

export enum ExamEventToggleFilter {
  Upcoming = 'upcoming',
  Passed = 'passed',
}

export enum HeaderNavTab {
  ExamEvents = 'examEvents',
}

export enum UIMode {
  Edit = 'edit',
  View = 'view',
}

export enum EnrollmentStatus {
  PAID,
  EXPECTING_PAYMENT,
  QUEUED,
  CANCELED,
}
