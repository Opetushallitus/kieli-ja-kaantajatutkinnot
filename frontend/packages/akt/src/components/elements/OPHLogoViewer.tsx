import { FC } from 'react';

import { Svg } from 'components/elements/Svg';
import { getCurrentLang, useCommonTranslation } from 'configs/i18n';
import { AppLanguage, Direction } from 'enums/app';
import OPHLogoENHorizontal from 'public/assets/svg/oph_logo_horiz_en.svg';
import OPHLogoHorizontal from 'public/assets/svg/oph_logo_horiz_fi_sv.svg';
import OPHLogoENVertical from 'public/assets/svg/oph_logo_vert_en.svg';
import OPHLogoVertical from 'public/assets/svg/oph_logo_vert_fi_sv.svg';

interface OPHLogoViewerProps {
  direction: Direction;
  className?: string;
}

export const OPHLogoViewer: FC<OPHLogoViewerProps> = ({
  className,
  direction,
}) => {
  const currentLang = getCurrentLang();
  const translateCommon = useCommonTranslation();

  const getLogo = () => {
    const isEnglish = currentLang === AppLanguage.English;

    if (isEnglish && direction === Direction.Horizontal) {
      return OPHLogoENHorizontal;
    } else if (isEnglish && direction === Direction.Vertical) {
      return OPHLogoENVertical;
    } else if (!isEnglish && direction === Direction.Horizontal) {
      return OPHLogoHorizontal;
    } else if (!isEnglish && direction === Direction.Vertical) {
      return OPHLogoVertical;
    }
  };

  return (
    <Svg
      className={className}
      src={getLogo()}
      alt={translateCommon('ophLogo')}
    />
  );
};
