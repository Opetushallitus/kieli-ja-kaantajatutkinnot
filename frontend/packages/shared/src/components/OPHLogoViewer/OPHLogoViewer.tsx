import { FC } from 'react';

import { AppLanguage, Direction } from '../../enums/common';
import OPHLogoENHorizontal from '../../static/svg/oph_logo_horiz_en.svg';
import OPHLogoHorizontal from '../../static/svg/oph_logo_horiz_fi_sv.svg';
import OPHLogoENVertical from '../../static/svg/oph_logo_vert_en.svg';
import OPHLogoVertical from '../../static/svg/oph_logo_vert_fi_sv.svg';
import { Svg } from '../Svg/Svg';

interface OPHLogoViewerProps {
  direction: Direction;
  className?: string;
  alt: string;
  currentLang: string;
}

export const OPHLogoViewer: FC<OPHLogoViewerProps> = ({
  className,
  direction,
  alt,
  currentLang,
}) => {
  const getLogo = () => {
    const isEnglish = currentLang === AppLanguage.English;

    if (isEnglish && direction === Direction.Horizontal) {
      return OPHLogoENHorizontal;
    } else if (isEnglish && direction === Direction.Vertical) {
      return OPHLogoENVertical;
    } else if (!isEnglish && direction === Direction.Horizontal) {
      return OPHLogoHorizontal;
    } else if (!isEnglish && direction === Direction.Vertical) {
      return OPHLogoVertical;
    }
  };

  return <Svg className={className} src={getLogo()} alt={alt} />;
};
