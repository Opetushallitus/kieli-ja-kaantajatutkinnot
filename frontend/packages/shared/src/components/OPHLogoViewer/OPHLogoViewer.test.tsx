import renderer from 'react-test-renderer';

import { AppLanguage, Direction } from '../../enums/app';
import { OPHLogoViewer } from './OPHLogoViewer';

describe('OPHLogoViewer', () => {
  it('should render OPHLogoViewer correctly', () => {
    const tree = renderer
      .create(
        <OPHLogoViewer
          direction={Direction.Horizontal}
          alt="Opetushallituksen logo"
          currentLang={AppLanguage.Finnish}
        />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
