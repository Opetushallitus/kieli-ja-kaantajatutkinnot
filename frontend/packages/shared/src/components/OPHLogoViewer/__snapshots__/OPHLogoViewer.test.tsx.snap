// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OPHLogoViewer should render OPHLogoViewer correctly 1`] = `
<img
  alt="Opetushallituksen logo"
  src="test"
/>
`;
